### PR TITLE
👕 refactor: Refactor handle_request method to improve robustness

### PR DESCRIPTION
**Pull request by Cyber-Freak999**
# Description
This PR refactors the `_handle_request` method in the BaseAPI class to improve reliability, exception management, and logging efficiency. The changes ensure that the API client behaves more predictably under failure conditions and performs better in logging operations.

# Details
- Added a timeout argument to the `requests.request` method to avoid indefinite hangs.
- Improved exception handling by capturing and raising specific exceptions (`ConnectTimeout`, `ConnectionError`) to provide clearer error feedback during network issues.
- Introduced lazy string interpolation using `%` formatting for logging within the `_handle_request` method to enhance performance and prevent premature string formatting.
- Enhanced the API response parsing method for cleaner handling of status codes.
- Fixed an issue with `_url()` method where it referenced an incorrect base URL variable.

### CODE REVIEW

**Comment by Cyber-Freak999** [comment]
Hey @phyleria
Please have a look at this

**Comment by phyleria** [comment]
Ok ✅

File: submissions/Chimoney-Python/chimoney/Base.py [hunk 1]
```diff
@@ -84,10 +119,17 @@ def _handle_request(self, method_type, path, data=None, params=None):
             dict: The response from the Chi Money API.
         """
 
-        if data:
-            payload = json.dumps(data)
-        else:
-            payload = None
+        # if data:
+        #     payload = json.dumps(data)
```
Comment (todak2000) [inline submissions/Chimoney-Python/chimoney/Base.py:123]: kindly remove redundant/commented out lines of code.

**Review by todak2000 — CHANGES_REQUESTED** [review]
Generally, it would be super cool if you could take the following into consideration
1.  For some errors, an API Response is returned, which is fine while for others, an exception is raised. Consider standardizing this behavior for consistency sake

2. The method is using `requests.request()` directly instead of the session object created in `__init__`. Using the session object could improve performance for multiple requests.

3. The retry logic for rate limiting (status code 429) is good, but consider implementing a maximum number of retries to prevent infinite loops.

File: submissions/Chimoney-Python/chimoney/Base.py (hunk 1 above)
Comment (Cyber-Freak999) [reply submissions/Chimoney-Python/chimoney/Base.py:123]: Will do

**Comment by Cyber-Freak999** [comment]
@phyleria @todak2000 
Have a look at the improvement I made

**Review by todak2000 — APPROVED** [review]

**Pull request merged**